modules/git/commit.go

```diff
@@ -264,23 +266,26 @@ func (c *Commit) CommitsBefore() (*list.List, error) {
 
 // HasPreviousCommit returns true if a given commitHash is contained in commit's parents
 func (c *Commit) HasPreviousCommit(commitHash SHA1) (bool, error) {
-	for i := 0; i < c.ParentCount(); i++ {
-		commit, err := c.Parent(i)
+	if err := CheckGitVersionAtLeast("1.8"); err != nil {
+		_, err := NewCommand("merge-base", "--ancestor", commitHash.String(), c.ID.String()).RunInDir(c.repo.Path)
```

Comment:
Shouldn't this be `--is-ancestor`? I don't think `--ancestor` is a valid argument.

Reply:
yup it should be - the is has been deleted through fat fingers!

```suggestion
		_, err := NewCommand("merge-base", "--is-ancestor", commitHash.String(), c.ID.String()).RunInDir(c.repo.Path)
```